shared/src/main/scala/amf/core/AMFGraphPartialCompiler.scala

```diff
@@ -0,0 +1,98 @@
+package amf.core
+
+import amf.client.remod.amfcore.plugins.parse.AMFParsePlugin
+import amf.client.remod.amfcore.plugins.{HighPriority, PluginPriority}
+import amf.core.errorhandling.AMFErrorHandler
+import amf.core.exception.UnsupportedParsedDocumentException
+import amf.core.metamodel.Obj
+import amf.core.metamodel.Type.ObjType
+import amf.core.metamodel.document.{BaseUnitModel, DocumentModel}
+import amf.core.metamodel.domain.DomainElementModel
+import amf.core.model.document.{BaseUnit, Document}
+import amf.core.model.domain.{AmfObject, DomainElement}
+import amf.core.parser.{Annotations, Fields, ParserContext, ReferenceHandler, SyamlParsedDocument}
+import amf.plugins.document.graph.parser.{
+  FlattenedGraphParser,
+  FlattenedUnitGraphParser,
+  GraphDependenciesReferenceHandler,
+  GraphParserContext
+}
+
+class AMFGraphPartialCompiler(compilerContext: CompilerContext, startingPoint: String)
+    extends AMFCompiler(compilerContext, Some("application/graph+json")) {
+
+  implicit val executionContext = compilerContext.parserContext.config.executionContext
+
+  private val parsePlugin = new AMFParsePlugin {
+    override def parse(document: Root, ctx: ParserContext): BaseUnit = {
+      document.parsed match {
+        case s: SyamlParsedDocument =>
+          val maybeObject =
+            new FlattenedGraphParser(ctx.config, startingPoint)(new GraphParserContext(eh = ctx.config.eh))
+              .parse(s.document)
+          val obj = maybeObject match {
+            case Some(obj) => obj
+            case _ =>
+              new DomainElement {
```

Comment:
Extract to class/method

Reply:
Is the only place where is build, I don't know if having it separately apports anything